### ISSUE TITLE
Convert IEnumerable<Result<T>> to Result<IEnumerable<T>>

### ISSUE THREAD

**Issue by RikDriever** [vkhorikov/CSharpFunctionalExtensions#269]
I have a ValueObject `Sample` with a static factory method `Of` that returns a `Result<Sample>`.

In a command handler I want to convert a collection of primitives to samples:

`var myCollection = request.PrimitiveValues.Select(x => Sample.Of(x));`
or shorter:
`var myCollection = request.PrimitiveValues.Select(Sample.Of);`

`myCollection` is now of type `IEnumerable<Result<Sample>>`.

In order to continue handling the command, I want to know if any of the results is a failure and if so, return a (ideally a combined) result.

If all results are a success, I want to continue with a collection of Samples.

So basically I want to convert `IEnumerable<Result<Sample>>` to `Result<IEnumerable<Sample>>`,  any pointers to guide me in the right direction?

Edit:
For now I solved it like below, but I feel there should be a better/more elegant way.

```
var results = request.PrimitiveValues.Select(Sample.Of).ToList();
var failures = results.Where(x => x.IsFailure).ToList();
if (failures.Any())
{
   return failures.Combine();
}
var samples = results.Select(x => x.Value).ToList();
```

**Comment by hankovich** [comment]
>but I feel there should be a better/more elegant way


You're absolutely right, it should! 
And it exists :)

```csharp
request.PrimitiveValues.Select(Sample.Of).Combine();
```

**Comment by RikDriever** [comment]
So simple, so elegant, I love it, thanks :)
